lib/sport_ngin_aws_auditor/scripts/audit.rb

```diff
@@ -35,9 +35,10 @@ def self.execute(environment, options=nil, global_options=nil)
 
         cycle.each do |c|
           all_data = gather_data(c.first, tag_name) if (c.last || no_selection)
-          data = all_data.first unless all_data.nil?
-          retired_reserved_instances = all_data.last if (all_data.length == 2 && !all_data.nil?)
-          print_data(slack, environment, data, retired_reserved_instances, c.first) if (c.last || no_selection)
+          data = all_data[0] unless all_data.nil?
+          retired_tags = all_data[1] if (all_data.length >= 2 && !all_data.nil?)
+          retired_ris = all_data[2] if (all_data.length >= 3 && !all_data.nil?)
```

Comment:
Encapsulating the data in an class would make this section much more readable and concise as well.